math.go

```diff
@@ -0,0 +1,139 @@
+package goscale
+
+import (
+	"errors"
+	"math"
+	"math/bits"
+)
+
+var (
+	errOverflow  = errors.New("overflow")
+	errunderflow = errors.New("underflow")
+)
+
+func Clamp(value, min, max int) int {
+	if value < min {
+		return min
+	}
+	if value > max {
+		return max
+	}
+
+	return value
+}
+
+func MaxU64(a, b U64) U64 {
```

Comment:
The naming could be improved as it's currently somewhat inconsistent, in one place, it's `MaxU<T>(a, b)`, while in another it's `Max<T>(a, b)` without the U/I. 
Also, `Max<T>(a, b)`, which compares two values, and `Max<T>()`, which returns the maximum value from a range, appear too similar.

Reply:
You cannot have MaxU64(a,b) and MaxU64 at the same time. The case is for Max128, MaxU128 and MaxI128 as we export the constants as well.

Reply:
yes, we don't need to have `MaxU<T>(a,b)` and `MaxU<T>` since they do completely different things. My point was that their names look very similar, so maybe we need to differentiate them a bit

* `MaxU128(a, b)`, `MinU128(a, b)`
* `U128Max()`, `U128Min()`